ui/server/src/main/scala/pl/touk/nussknacker/ui/process/processingtypedata/ProcessingTypeDataReload.scala

```diff
@@ -38,14 +44,23 @@ class BasicProcessingTypeDataReload(loadMethod: () => ProcessingTypeDataProvider
 
 object BasicProcessingTypeDataReload {
 
-  def wrapWithReloader(loadMethod: () => ProcessingTypeDataProvider[ProcessingTypeData]): (ProcessingTypeDataProvider[ProcessingTypeData], ProcessingTypeDataReload) = {
-    val reloader = new BasicProcessingTypeDataReload(loadMethod)
+  def wrapWithReloader(loadMethod: () => ProcessingTypeDataProvider[ProcessingTypeData]): (ProcessingTypeDataProvider[ProcessingTypeData], ProcessingTypeDataReload with Initialization) = {
+    // must be lazy to avoid problems with dependency injection cycle - see NusskanckerDefaultAppRouter.create
+    lazy val reloader = new BasicProcessingTypeDataReload(loadMethod)
     val provider = new ProcessingTypeDataProvider[ProcessingTypeData] {
       override def forType(typ: ProcessingType): Option[ProcessingTypeData] = reloader.current.forType(typ)
 
       override def all: Map[ProcessingType, ProcessingTypeData] = reloader.current.all
     }
-    (provider, reloader)
+    (provider, new ProcessingTypeDataReload with Initialization {
+      override def init(): Unit = {
+        reloader
+      }
+
+      override def reloadAll(): Unit = {
```

Comment:
syntax without braces would be clearer just for the sake of composition of this class

Reply:
what do u mean? to remove `{ }`? how it is related to composition?